pkg/asset/installconfig/azure/validation.go

```diff
@@ -119,6 +120,47 @@ func validateRegion(client API, fieldPath *field.Path, p *aztypes.Platform) fiel
 	return field.ErrorList{field.Invalid(fieldPath, p.Region, fmt.Sprintf("region %q does not support resource creation", p.Region))}
 }
 
+// validateRegionForUltraDisks checks that the Ultra SSD disks are available for the user.
+func validateRegionForUltraDisks(fldPath *field.Path, region string) field.ErrorList {
+	allErrs := field.ErrorList{}
+	diskType := "UltraSSD_LRS"
+
+	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer cancel()
+	client, err := NewClient(ctx)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath.Child("disktype"), err))
+	}
+
+	contxt, canc := context.WithTimeout(context.TODO(), 30*time.Second)
+	defer canc()
+
+	regionDisks, err := client.GetDiskSkus(contxt, region)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath.Child("disktype"), err))
+	} else {
+		flag := true
+
+		for _, page := range regionDisks {
+			for _, location := range to.StringSlice(page.Locations) {
+				if location == diskType {
+					flag = false
+					break
```

Comment:
instead of using `flag` can't you just return allErrs here? this should help simplify.